WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt

```diff
@@ -98,6 +104,14 @@ class ReaderPostUiStateBuilder @Inject constructor(
         )
     }
 
+    fun mapTagListToReaderInterestUiState(interests: ReaderTagList, onClicked: ((String) -> Unit)) =
+            ReaderInterestCardUiState(interests.take(READER_INTERESTS_LIST_SIZE).map { interest ->
+                ReaderInterestUiState(
+                        interest.tagTitle,
+                        interests.indexOf(interest) != READER_INTERESTS_LAST_INDEX, onClicked
+                )
+            })
```

Comment:
So for this approach here, the idea is that we need to have 5 options for the topics the user may like so I simply took the first five topics and I utilize the current index to determine if the `uiState` should have a divider since we don't want the last item of the interest card list to show a divider. Let me know if this approach is fine or if it can be simplified or improved in anyway. I have the last index hardcoded and I was wondering if I should create a variable to store the result of `interest.take(5)` and then utilize that to determine the index. So yes, any feedback or improvement here is welcome. 

Also let me know if this is the right place for this type of business logic. I was wondering if this should be from the data layer. I put it here because this is somewhat view logic, so let me know that as well.

Reply:
Few observations:

1.  If there are less than 5 elements - let's say total of 3, `interests.indexOf(interest) != READER_INTERESTS_LAST_INDEX`  will be true for the last element at index 2 and divider will be displayed for the last element. wdyt?
2. `onClicked` can be moved to a new line (each param on separate line).
3.  I don't think it is necessary to put visibility logic inside the VM. We can create a `buildDividerVisibility` fn similar to the `buildPhotoFrameVisibility` inside `ReaderPostUiState` for consistency.

Reply:
Great observations :) 6409d7f